### ISSUE TITLE
How to properly launch robotlab on OSX?

### ISSUE THREAD

**Issue by datakurre** [robots-from-jupyter/robotlab#7]
Today @ju55i tested robotlab installer https://dev.azure.com/nickbollweg/nickbollweg/_build/results?buildId=99 on OSX

Only issue was that the proper way to launch the robotlab was unclear.

Installer prompts about updating bashrc, but that could be something that not everyone wants. Besides that Jussi was using zsh and the source-command proposed by the installer did not update PATH (so neither robotlab nor geckodriver/chromedriver was on path).

After manually setting PATH to robotlab/bin everything worked as expected.

**Comment by bollwyvl** [comment]
Ha, yeah: still getting everything to _work_, but it indeed needs to work _well_ and not require referring to the docs. That's why I hadn't started advertising the builds. Thanks for trying it out!

On windows, we can actually create start menu items, so having a "RobotLab" and a "RobotLab Command Line" would be reasonable. Historically, this also worked on linux and osx, but no longer does, apparently.

On the other platforms, we can run one script (bat/sh) at the end of a successful install that knows where it was installed. This could just be some docs, like:

```bash
Thanks for installing RobotLab 0.7.1! 

To use it first _activate_ the environment:

  . [/path/to/robotlab/]bin/activate

Once activated, try one of the robotlab commands

  robotlab             # start the pre-bundled JupyterLab
  robotlab-extensions  # add/remove extensions to JupyterLab
  nbrobot              # run robot tests, but also include robotkernel ipynb files
  robotlab-examples    # install robotkernel examples into the current directory

To uninstall RobotLab, just delete the directory

  rm -rf /path/to/robotlab
```

Or something more interactive:
```bash
Thanks for installing RobotLab!

[V]iew documentation
[C]reate a RobotLab and RobotLab Command Line desktop shortcut
[R]un RobotLab now

What would you like to do? [V/c/r]
```

All the way up to a tcl dialog (weird). Heck, I think we have qt in the build, so really the sky's the limit. However, the whole thing is run as a _user_, not an _administrator_, so there will be some things we can't guarantee being able to do.

**Comment by bollwyvl** [comment]
Making some progress here... with [pyshortcuts](https://github.com/newville/pyshortcuts) it's easy to launch any python module, though a little more complicated to set the environment properly (e.g. for *driver). Much like a "regular" conda install, i wanted to have a "shell" icon which would launch a native terminal... this a bit harder. Anyhow, hopefully will have something figured out (and tested!) soon!

**Comment by bollwyvl** [comment]
Ok, with #14 I've added a desktop shortcut. I've tested them locally on linux (it doesn't work with my gnome 3, meh) and windows. I know the `RobotLab.app` is getting created on the OSX desktop, but haven't clicked on one myself, so I don't know if it _really_ works!

**Comment by bollwyvl** [comment]
Working on proper releases on github (#16), but here are the most recent builds after #15:
- [linux](https://dev.azure.com/robots-from-jupyter/1e5f1f30-0c52-4d28-8cf7-8c72252fbb49/_apis/build/builds/32/artifacts?artifactName=RobotLab%20for%20Linux&api-version=5.0&%24format=zip)
- [osx](https://dev.azure.com/robots-from-jupyter/1e5f1f30-0c52-4d28-8cf7-8c72252fbb49/_apis/build/builds/32/artifacts?artifactName=RobotLab%20for%20MacOSX&api-version=5.0&%24format=zip)
- [win](https://dev.azure.com/robots-from-jupyter/1e5f1f30-0c52-4d28-8cf7-8c72252fbb49/_apis/build/builds/32/artifacts?artifactName=RobotLab%20for%20Windows&api-version=5.0&%24format=zip)
  - note, i found an issue with the shortcut, working on it now (#17).

**Comment by datakurre** [comment]
I'm currently testing OSX one more time (build 32) and try to figure out what's broken there.

**Comment by datakurre** [comment]
This is the remaining issue on OSX:

```
/Users/datakurre/robotlab/python.app/Contents/MacOS/python /Users/datakurre/robotlab/lib/python3.6/site-packages/robotlab/launch.py
➜ ~ /Users/datakurre/robotlab/python.app/Contents/MacOS/python /Users/datakurre/robotlab/lib/python3.6/site-packages/robotlab/launch.py
zsh: no such file or directory: /Users/datakurre/robotlab/python.app/Contents/MacOS/python
```

I had to patch `~/Desktop/RobotLab.app/Contents/MacOS/RobotLab` with

```diff
--- a	2019-01-14 18:40:04.000000000 +0200
+++ b	2019-01-14 18:39:43.000000000 +0200
@@ -1,7 +1,7 @@
 #!/bin/bash
 ## Run script with Python that created this script
 export PYTHONEXECUTABLE=/Users/datakurre/robotlab/bin/python
-export PY=/Users/datakurre/robotlab/python.app/Contents/MacOS/python
+export PY=/Users/datakurre/robotlab/bin/python
 export SCRIPT=/Users/datakurre/robotlab/lib/python3.6/site-packages/robotlab/launch.py
 export ARGS=''                            
```                                                                               

To make this work.

**Comment by bollwyvl** [comment]
Hm. we could ship python.app:
https://anaconda.org/anaconda/python.app
So it does exist.

**Comment by bollwyvl** [comment]
https://github.com/newville/pyshortcuts/blob/master/pyshortcuts/darwin.py#L49

**Comment by bollwyvl** [comment]
actually, monkeypatch is probably easier.

**Comment by datakurre** [comment]
@bollwyvl Whatever :) If python.app does something fancy, you could make a pull for it and give a link so that I can try it out later today (like in 2 hours or so).

**Comment by bollwyvl** [comment]
yeah, i'll have something up before then.

**Comment by datakurre** [comment]
Meanwhile I should repeat how awesome this is (and how crazy the sh-script based packaging is).

**Comment by bollwyvl** [comment]
lol yeah. **Luckily** i have complete confidence in its craziness.

**Comment by bollwyvl** [comment]
Builds will be here _provided_ everything works as it does locally.
https://dev.azure.com/robots-from-jupyter/robots-from-jupyter/_build/results?buildId=36&view=results

**Comment by bollwyvl** [comment]
Closed by #18
